clients/src/main/java/org/apache/kafka/clients/ClientResponseWithFinalize.java

```diff
@@ -0,0 +1,26 @@
+package org.apache.kafka.clients;
+
+import java.nio.ByteBuffer;
+import org.apache.kafka.common.errors.AuthenticationException;
+import org.apache.kafka.common.errors.UnsupportedVersionException;
+import org.apache.kafka.common.memory.MemoryPool;
+import org.apache.kafka.common.requests.AbstractResponse;
+import org.apache.kafka.common.requests.RequestHeader;
+import org.apache.kafka.common.utils.LogContext;
+
+
+public class ClientResponseWithFinalize extends ClientResponse {
```

Comment:
javadoc - "this is a decorator for ClientResponse used to verify (at finalization time) that any underlying memory for this response has been returned to the pool. to be used only as a debugging aide"